src/aatoolbox/datasources/ecmwf/api_seas5_monthly.py

```diff
@@ -0,0 +1,288 @@
+https://www.ecmwf.int/en/forecasts/access-forecasts/ecmwf-web-api
+"""
+
+import logging
+from datetime import date
+from pathlib import Path
+from typing import Union
+
+import geopandas as gpd
+import pandas as pd
+import xarray as xr
+from ecmwfapi import ECMWFService
+from ecmwfapi.api import APIException
+
+from aatoolbox.datasources.datasource import DataSource
+from aatoolbox.utils.geoboundingbox import GeoBoundingBox
+from aatoolbox.utils.io import check_file_existence
+
+logger = logging.getLogger(__name__)
+_MODULE_BASENAME = "ecmwf"
+# folder structure within the ecmwf dir
+SEAS_DIR = "seasonal-monthly-individual-members"
+PRATE_DIR = "prate"
+GRID_RESOLUTION = 0.4  # degrees
+
+
+class EcmwfApi(DataSource):
+    """
+    Work with ECMWF's API data.
+
+    Parameters
+    ----------
+    iso3 : (str)
+        country iso3
+    """
+
+    def __init__(self, iso3: str, geobb):
+        super().__init__(
+            iso3=iso3, module_base_dir=_MODULE_BASENAME, is_public=False
+        )
+
+        # the geobb indicates the boundaries for which data is
+        # downloaded and processed
+        if type(geobb) == GeoBoundingBox:
+            self._geobb = geobb
+        elif type(geobb) == gpd.GeoDataFrame:
+            self._geobb = GeoBoundingBox.from_shape(geobb)
+        else:
+            self._geobb = GeoBoundingBox(north=90, south=-90, east=0, west=360)
+        # round coordinates to correspond with the grid ecmwf publishes
+        # its data on
+        self._geobb.round_coords(round_val=GRID_RESOLUTION)
+
+    def download(
+        self,
+        min_date: Union[str, date] = None,
+        max_date: Union[str, date] = None,
+        clobber: bool = False,
+        grid: float = GRID_RESOLUTION,
+    ):
+        """
+        Download the seasonal forecast precipitation by ECMWF from its API.
+
+        From the ECMWF API retrieve the mean forecasted monthly precipitation
+        per ensemble member
+
+        Parameters
+        ----------
+        min_date : Union[str,date], default = None
+            The first date to download data for.
+            If None, set to the first available date
+        max_date : Union[str,date], default = None
+            The last date to download dat for.
+            If None, set to the last available date
+            All dates between min_date and max_date are downloaded
+        clobber: bool, default = False
+            If True, overwrite downloaded files if they already exist
+        grid: float, default = 0.4
+            Grid resolution in degrees
+
+        Examples
+        --------
+        >>> import geopandas as gpd
+        >>> df_admin_boundaries = gpd.read_file(gpd.datasets.get_path('nybb'))
+        >>> download(iso3="nybb",ecmwf_dir=ecmwf_dir,
+        ... iso3_gdf=df_admin_boundaries.to_crs("epsg:4326"))
+        """
+        if min_date is None:
+            min_date = "1992-01-01"
+        if max_date is None:
+            max_date = date.today().replace(day=1)
```

Comment:
Hmm I could see this issue with processing but not with downloading. As it is just downloading all the files that are available. And in the downloaded filename the date is included. 
Then we process the results into one dataset. This merges all the available files. But maybe in the metadata it could be included what the last date is that the processed dataset is updated. Is that what you mean?

Reply:
Yes I meant it mainly for the processing. And for example, the metadata could even contain a list of files that were used to construct the final product, as well as the dates.

Reply:
I now added all filenames to an attr `included_files`. It is a huge list but think it makes sense. What do you think?

Reply:
I love it!